### PR TITLE
Error message minor

### PR DESCRIPTION
**Pull request by juancarlospaco**
New users try to use typeless parameters and the warning says:

`"typeless parameters are now obsolete"`

Changed it to:

`"typeless parameters are not allowed, except for 'template'; See 'using' statement instead"`

Make the 2 existing messages consistent with each other.

The "... are obsolete" makes it looks like was just recently deprecated.

Instead it recommends users to learn about `using` https://nim-lang.org/docs/manual.html#statements-and-expressions-using-statement

Unrelated CI Fail, Bitbucket is 500'ing.

### CODE REVIEW

File: compiler/lineinfos.nim [hunk 1]
```diff
@@ -110,7 +110,7 @@ const
     warnFieldXNotSupported: "field '$1' not supported",
     warnRstStyle: "RST style: $1",
     warnCommentXIgnored: "comment '$1' ignored",
-    warnTypelessParam: "'$1' has no type. Typeless parameters are deprecated; only allowed for 'template'",
+    warnTypelessParam: "'$1' has no type. Typeless parameters are not allowed, except for 'template'; See 'using' statement instead",
```
Comment (timotheecour) [inline compiler/lineinfos.nim]: dead code! we should remove it, 
```suggestion
    warnTypelessParam: "", # deadcode
```
(and can be removed in another PR)

File: compiler/semtypes.nim [hunk 2]
```diff
@@ -1297,7 +1297,7 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
         let param = strTableGet(c.signatures, arg.name)
         if param != nil: typ = param.typ
         else:
-          localError(c.config, a.info, "typeless parameters are obsolete")
+          localError(c.config, a.info, "typeless parameters are not allowed, except for 'template'; See 'using' statement instead")
```
Comment (timotheecour) [inline compiler/semtypes.nim]:
```suggestion
          localError(c.config, a.info, "typeless parameters are not allowed, except with 'using' or for 'template|macro|generic' parameters")
```

how about this?

Comment (juancarlospaco) [reply compiler/semtypes.nim]: Wait, AFAIK Generic has types, are generic but still types ?. :thinking:  Generic types are not typeless.

Comment (timotheecour) [reply compiler/semtypes.nim]:
```nim
proc fn[T]() # typeless param
proc fn[T: enum]() # param with type class specification
```
that's what i mean; but maybe ok to remove `|generic` if unclear

Comment (juancarlospaco) [reply compiler/semtypes.nim]: Oh ok. :+1:

File: compiler/semtypes.nim [hunk 3]
```diff
@@ -1297,7 +1297,7 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
         let param = strTableGet(c.signatures, arg.name)
         if param != nil: typ = param.typ
         else:
-          localError(c.config, a.info, "typeless parameters are obsolete")
+          localError(c.config, a.info, "typeless parameters are not allowed, except with 'using' or for 'template|macro|generic' parameters")
```
Comment (ringabout) [inline compiler/semtypes.nim]:
```suggestion
          localError(c.config, a.info, "typeless parameters are not allowed, except with 'using' or for 'templates|macros|generics parameters'")
```

File: compiler/semtypes.nim [hunk 4]
```diff
@@ -1297,7 +1297,7 @@ proc semProcTypeNode(c: PContext, n, genericParams: PNode,
         let param = strTableGet(c.signatures, arg.name)
         if param != nil: typ = param.typ
         else:
-          localError(c.config, a.info, "typeless parameters are obsolete")
+          localError(c.config, a.info, "typeless parameters are not allowed, except with 'using' or for 'templates|macros|generics parameters'")
```
Comment (Araq) [inline compiler/semtypes.nim]:
```suggestion
          localError(c.config, a.info, "parameter '$1' requires a type" % param.name.s)
```
